core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCAdaptorPointComputer.scala

```diff
@@ -0,0 +1,101 @@
+package org.bitcoins.core.protocol.dlc
+
+import org.bitcoins.core.protocol.tlv.{
+  EnumOutcome,
+  SignedNumericOutcome,
+  UnsignedNumericOutcome
+}
+import org.bitcoins.crypto.{
+  CryptoUtil,
+  ECPublicKey,
+  FieldElement,
+  SchnorrPublicKey
+}
+import scodec.bits.ByteVector
+
+/** Responsible for optimized computation of DLC adaptor point batches. */
+object DLCAdaptorPointComputer {
+
+  private val base: Int = 2
+
+  /** Computes:
+    *     nonce + outcomeHash*pubKey
+    * where outcomeHash is as specified in the DLC spec.
+    */
+  def computePoint(
+      pubKey: SchnorrPublicKey,
+      nonce: ECPublicKey,
+      outcome: ByteVector): ECPublicKey = {
+    val hash = CryptoUtil
+      .sha256SchnorrChallenge(
+        nonce.schnorrNonce.bytes ++ pubKey.bytes ++ CryptoUtil
+          .sha256DLCAttestation(outcome)
+          .bytes)
+      .bytes
+
+    nonce.add(pubKey.publicKey.tweakMultiply(FieldElement(hash)))
+  }
+
+  /** Efficiently computes all adaptor points, in order, for a given ContractInfo.
+    * @see https://medium.com/crypto-garage/optimizing-numeric-outcome-dlc-creation-6d6091ac0e47
+    */
+  def computeAdaptorPoints(contractInfo: ContractInfo): Vector[ECPublicKey] = {
+    // The possible messages a single nonce may be used to sign
+    val possibleOutcomes: Vector[ByteVector] =
+      contractInfo.contractDescriptor match {
+        case enum: EnumContractDescriptor =>
+          enum.keys.map(_.outcome).map(CryptoUtil.serializeForHash)
+        case _: NumericContractDescriptor =>
+          0
+            .until(base)
+            .toVector
+            .map(_.toString)
+            .map(CryptoUtil.serializeForHash)
+      }
+
+    // Oracle -> Nonce -> Outcome -> SubSigPoint
+    // These are the points that are then combined to construct aggregate points.
+    val preComputeTable: Vector[Vector[Vector[ECPublicKey]]] =
```

Comment:
it might be worth adding a type for this? I find it hard to grasp conceptually what a `Vector[Vector[Vector[ECPublicKey]]]` is. Field names could be useful.

Reply:
it is just one thing so there wouldn't be any field names, this is just a three dimensional array of points, there's not much more I can imagine doing beyond the documentation above it rn

Reply:
Yeah i guess I interpreted your documentation to be a function signature (takes in 3 parameters, returns a `SubSigPoint`)